plane/src/drone/backend_manager.rs

```diff
@@ -262,6 +262,18 @@ impl<R: Runtime> BackendManager<R> {
             .expect("State lock is poisoned")
             .state
             .clone();
+
+        if let BackendState::Terminating { termination, .. } = state {
+            if termination == TerminationKind::Hard {
+                // We are already hard-terminating, don't change the kind.
+                return Ok(());
+            }
+            if termination == TerminationKind::Soft && kind == TerminationKind::Soft {
+                // We are already soft-terminating, don't change the kind.
+                return Ok(());
+            }
```

Comment:
We currently already have a check in `to_terminating` that makes sure a hard terminate can't turn into a soft terminate. Maybe it makes sense to protect against _sending_ more than one soft terminate, but I'm worried about dropping out here on a hard terminate. One reason is that the `set_state` call is one spot where we can actually detect that a hard terminating backend no longer exists and mark it as terminated. If a backend reaches a hard terminating state but we miss the event from docker to mark it as terminated, then I don't see a way the drone can ever mark the backend as terminated if we're dropping out here.

Reply:
Hmm good point, I'll mark this as a draft and have another think through it.